components/domains/steps/userInfoForm.tsx

```diff
@@ -0,0 +1,174 @@
+                helperText="Secure your domain's future and stay ahead with vital updates. Your email stays private with us, always."
+                label="Email address"
+                value={email}
+                onChange={(e) => changeEmail(e.target.value)}
+                color="secondary"
+                error={emailError}
+                errorMessage={"Please enter a valid email address"}
+                type="email"
+              />
+            ) : null}
+
+            {formState.needMetadata ? (
+              <SwissForm
+                isSwissResident={formState.isSwissResident}
+                onSwissResidentChange={() =>
+                  updateFormState({
+                    isSwissResident: !formState.isSwissResident,
+                  })
+                }
+              />
+            ) : null}
+            {type === FormType.REGISTER ? (
+              <SelectIdentity
+                tokenId={formState.tokenId}
+                changeTokenId={changeTokenId}
+              />
+            ) : null}
+            <NumberTextField
+              value={formState.duration}
+              label="Years to register (max 25 years)"
+              placeholder="years"
+              onChange={(e) => changeDuration(Number(e.target.value))}
+              incrementValue={() =>
+                changeDuration(
+                  formState.duration < maxYearsToRegister
+                    ? formState.duration + 1
+                    : formState.duration
+                )
+              }
+              decrementValue={() =>
+                changeDuration(
+                  formState.duration > 1
+                    ? formState.duration - 1
+                    : formState.duration
+                )
+              }
+              color="secondary"
+              required
+            />
+            {type === FormType.RENEW ? (
+              <RenewalDomainsBox
+                helperText="Check the box of the domains you want to renew"
+                setSelectedDomains={setSelectedDomains}
+                selectedDomains={formState.selectedDomains}
+              />
+            ) : null}
+          </div>
+        </div>
+        <div className={styles.summary}>
+          <div>
+            {address ? (
+              <Button
+                onClick={goToNextStep}
+                disabled={
+                  !formState.duration ||
+                  formState.duration < 1 ||
+                  (formState.needMetadata && emailError) ||
+                  (type === FormType.RENEW &&
+                    !areDomainSelected(formState.selectedDomains))
+                }
+              >
+                {formState.needMetadata && emailError
+                  ? "Enter a valid Email"
+                  : type === FormType.RENEW &&
+                    !areDomainSelected(formState.selectedDomains)
+                  ? "Select a domain to renew"
+                  : "Next step"}
+              </Button>
```

Comment:
to make the inline html less complex could we add the message and the disabling of this button in a const ?